gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc

```diff
@@ -60,43 +60,50 @@ int main(int argc, char* argv[]) {
     return 1;
   }
 
-  std::string initialRegValStr;
+  std::string initial_reg_val_str =
+      gematria::ConvertHexToString(kInitialRegVal);
+  std::string initial_mem_val_str =
+      gematria::ConvertHexToString(kInitialMemVal);
   {
-    // initialize initialRegValStr with kInitialRegValue in hex
-    std::stringstream ss;
-    ss << std::hex << kInitialRegVal << std::dec;
-    ss >> initialRegValStr;
-  }
-  std::string initialMemValStr;
-  {
-    // initialize initialMemValStr with kInitialMemValue in hex
+    // initialize initial_mem_val_str with kInitialMemValue in hex
     std::stringstream ss;
     ss << std::hex << kInitialMemVal << std::dec;
-    ss >> initialMemValStr;
+    ss >> initial_mem_val_str;
   }
   std::string register_defs_lines;
   const std::unique_ptr<gematria::LlvmArchitectureSupport> llvm_support =
       gematria::LlvmArchitectureSupport::X86_64();
-  const llvm::MCRegisterInfo& MRI = llvm_support->mc_register_info();
+  const llvm::MCRegisterInfo& reg_info = llvm_support->mc_register_info();
 
   // Iterate through all general purpose registers and vector registers
   // and add them to the register definitions.
   // TODO(9Temptest): Change GR64_NOREXRegClassID to GR64_NOREX2RegClassID when
   // the LLVM version is bumped to avoid including the new APX GPRs (r16-r31)
   // that have recently been added to LLVM.
   for (unsigned i = 0;
-       i < MRI.getRegClass(llvm::X86::GR64_NOREXRegClassID).getNumRegs(); ++i) {
-    llvm::StringRef reg_name = MRI.getName(
-        MRI.getRegClass(llvm::X86::GR64_NOREXRegClassID).getRegister(i));
-    register_defs_lines += std::string(kRegDefPrefix) + std::string(reg_name) +
-                           " " + initialRegValStr + "\n";
+       i < reg_info.getRegClass(llvm::X86::GR64_NOREXRegClassID).getNumRegs();
+       ++i) {
+    llvm::StringRef reg_name = reg_info.getName(
+        reg_info.getRegClass(llvm::X86::GR64_NOREXRegClassID).getRegister(i));
+    register_defs_lines = llvm::Twine(register_defs_lines)
```

Comment:
This will still make mroe copying, I'd keep the `+=` here, but use a twine for the concats, e.g.
```
register_defs_lines += llvm::Twine(kRegDefPrefix).concat(reg_name).concat(" ") ...;
```